internal/librarian/generate_e2e_test.go

```diff
@@ -0,0 +1,239 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build e2e
+// +build e2e
+
+package librarian
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/go-git/go-billy/v5/osfs"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing/cache"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/go-git/go-git/v5/storage/filesystem"
+	"github.com/googleapis/librarian/internal/config"
+)
+
+const (
+	localRepoDir       = "../../testdata/e2e/generate/repo"
+	localRepoBackupDir = "../../testdata/e2e/generate/repo_backup"
+)
+
+func TestRunGenerate(t *testing.T) {
+	t.Parallel()
+	rand.Seed(time.Now().UnixNano())
+	for _, test := range []struct {
+		name    string
+		cfg     *config.Config
+		wantErr bool
+	}{
+		{
+			name: "testRunSuccess",
+			cfg: &config.Config{
+				API:      "google/cloud/pubsub/v1",
+				Repo:     fmt.Sprintf("%s-%d", localRepoDir, rand.Intn(10000)),
+				Source:   "../../testdata/e2e/generate/api_root",
+				WorkRoot: filepath.Join(os.TempDir(), fmt.Sprintf("librarian-%d", rand.Intn(10000))),
+			},
+		},
+		{
+			name: "testRunFailed",
+			cfg: &config.Config{
+				API:      "google/invalid/path",
+				Repo:     fmt.Sprintf("%s-%d", localRepoDir, rand.Intn(10000)),
+				Source:   "../../testdata/e2e/generate/api_root",
+				WorkRoot: filepath.Join(os.TempDir(), fmt.Sprintf("librarian-%d", rand.Intn(10000))),
+			},
+			wantErr: true,
```

Comment:
Can/should we have a more specific error check? (e.g. error message/type). If we just check to see if there is an error or not, we may not be confident in what is failing.

Reply:
In go/librarian:generate-command, there's no error types/return code defined for generate command.

If the generation failed, we can expect the `generate-response.json` contains a `error` field.